bindings/go/descriptor/normalisation/json/v4alpha1/normalisation.go

```diff
@@ -151,13 +151,23 @@ func DefaultComponent(d *v2.Descriptor) {
 // ProviderAsMap converts a provider string into a map structure if possible.
 // This function is used during normalization to handle provider information
 // in a standardized way.
-func ProviderAsMap(v interface{}) interface{} {
-	var provider map[string]interface{}
-	err := json.Unmarshal([]byte(v.(string)), &provider)
-	if err == nil {
-		return provider
+// ProviderAsMap tries to parse a JSON-encoded provider string into a map.
+func ProviderAsMap(v any) any {
+	switch v := v.(type) {
+	case []byte:
+		var provider map[string]any
+		if err := json.Unmarshal(v, &provider); err == nil {
+			return provider
+		}
+	case string:
+		var provider map[string]any
+		if err := json.Unmarshal([]byte(v), &provider); err == nil {
+			return provider
+		}
+	}
+	return map[string]any{
+		"name": v,
 	}
```

Comment:
```suggestion
func ProviderAsMap(v any) any {
    var provider map[string]any
	switch v := v.(type) {
	case []byte:
		if err := json.Unmarshal(v, &provider); err == nil {
			return provider
		}
	case string:
		if err := json.Unmarshal([]byte(v), &provider); err == nil {
			return provider
		}
	}
	return map[string]any{
		"name": v,
	}
```

Does the caller expect the return type to be `any` here?